docs/ref/models/querysets.txt

```diff
@@ -1664,7 +1664,8 @@ For example::
 
 All matched entries will be locked until the end of the transaction block,
 meaning that other transactions will be prevented from changing or acquiring
-locks on them.
+locks on them. Note that the locks are acquired when the QuerySet is evaluated,
+the example above *did not lock the rows yet.*
```

Comment:
This is in fact true which makes the example not very good. 

The problem with the example is not only that the rows are not locked because the queryset was not evaluated yet, but that it's not executed in a transaction block. 

Maybe the example should also include a code to open a transaction:

```python

with transaction.atomic():
    entries = Entry.objects.select_for_update().filter(author=request.user)
```

Reply:
To make it more explicit, we could also break down the example to show when the rows are actually locked:

```python
# The queryset hasn't been evaluated, so nothing is locked yet.
entry_queryset = Entry.objects.select_for_update().filter(author=request.user)
with transaction.atomic():
    # When iterating, the queryset is evaluated and the lock is acquired.
    for entry in entry_queryset:
        if entry.author == 'Foo':
            entry.author = 'Bar'
            entry.save()
```